dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1002 @@
+        # created. The automatically created disk has the same name as the
+        # gce_instance
+        if disk_resource.name == resource.name:
+          disk_resource.creation_timestamp = resource.creation_timestamp
+          resource.parent = disk_resource
+          disk_resource.id = resource.id + '-disk'
+
+        resource.disks.append(disk_resource)
+
+        if disk_resource.parent and not disk_resource.parent.id:
+          matched_resource = self._FindResource(
+              disk_resource.parent.name, disk_resource.parent.type,
+              disk_resource.parent.zone, disk_resource.parent.project_id)
+          if matched_resource:
+            disk_resource.parent = matched_resource
+
+        if disk_resource and not resource.parent:
+          resource.parent = disk_resource
+
+        if disk_resource.id:
+          self.resources_dict[disk_resource.id] = disk_resource
+
+    if resource.parent and not resource.parent.id:
+      matched_resource = self._FindResource(resource.parent.name,
+                                            resource.parent.type,
+                                            resource.parent.zone,
+                                            resource.parent.project_id)
+      if matched_resource:
+        resource.parent = matched_resource
+
+  def _BuildResourcesParentRelationships(self) -> None:
+    """Builds parent relationship for all resources."""
+    # Using resource_keys because self._resources_dict changes during the loop
+    resource_keys = list(self.resources_dict.keys())
+    for resource_key in resource_keys:
+      resource = self.resources_dict.get(resource_key)
+      if resource:
+        resource.parent = self._GetResourceParentTree(resource)
+
+  def _SearchForDeletedResource(
+      self, resource: Resource,
+      start_timestamp: datetime) -> Optional[Resource]:
+    """Searches for deleted resource in GCP Logs.
+
+    Args:
+      resource: resource to search for.
+      start_timestamp: the initial point of time to start the search
+
+    Return:
+      Found resource or None
+    """
+    if not resource or not start_timestamp:
+      return None
+
+    if resource.project_id != self.project_id:
+      return None
+
+    while start_timestamp > (datetime.now(timezone.utc) - timedelta(days=400)):
+
+      end_timestamp = start_timestamp + timedelta(minutes=20)
+      start_timestamp = start_timestamp - timedelta(days=30)
+
+      log_messages = self._GetLogMessages(resource.project_id, start_timestamp,
+                                          end_timestamp, resource.id)
+      self._ParseLogMessages(log_messages)
+
+      matched_resource = self._FindResource(resource.name, resource.type)
+      if matched_resource:
+        if matched_resource.deletion_timestamp and matched_resource.creation_timestamp:  # pylint: disable=line-too-long
+          return matched_resource
+
+    return None
+
+  def _RetrieveListOfDisks(self, project_id: str) -> Dict[str, Resource]:
```

Comment:
why don't you use this https://github.com/google/cloud-forensics-utils/blob/719093b4a229e5e97c30d93faabb1ccf3b6ee422/libcloudforensics/providers/gcp/internal/compute.py#L108

this adds code that wraps gcp APIs while we have libcloudforensics for that and it is used in dftimewolf for any cloud api operation